tests/tests/test-precompile-sacrifice.ts

```diff
@@ -0,0 +1,102 @@
+import { expect } from "chai";
+import { GENESIS_ACCOUNT, GENESIS_ACCOUNT_PRIVATE_KEY } from "../util/constants";
+import { customWeb3Request } from "../util/providers";
+import { describeDevMoonbeam } from "../util/setup-dev-tests";
+import { createContract } from "../util/transactions";
+
+describeDevMoonbeam("Precompiles - sacrifice", (context) => {
+  it("should be valid", async function () {
+    const txCall = await customWeb3Request(context.web3, "eth_call", [
+      {
+        from: GENESIS_ACCOUNT,
+        value: "0x0",
+        gas: "0x1000000",
+        gasPrice: "0x01",
+        to: "0x00000000000000000000000000000000000001FF",
+        data: `0x0000000000000000000000000000000000000000000000000000000000005BA0`, // 23456
+      },
+    ]);
+
+    console.log(txCall);
+
+    // should return empty result
+    expect(txCall.result).equals("0x");
+  });
+});
+
+describeDevMoonbeam("Precompiles - sacrifice", (context) => {
+  let contract: Contract;
+  let nonce = 0;
+
+  before("Setup: Deploy contract", async function () {
+    const result = await createContract(context.web3, "SacrificeWrapper");
+    await context.createBlock({ transactions: [result.rawTx] });
+    contract = result.contract;
+
+    nonce++;
+  });
+
+  // helper to send a txn to call sacrifice with a specified amount of gas.
+  async function transact(amount: Number) {
+    // create and sign txn...
+    const tx = await context.web3.eth.accounts.signTransaction(
+      {
+        from: GENESIS_ACCOUNT,
+        to: contract.options.address,
+        gas: "0x100000",
+        nonce: nonce++,
+        data: contract.methods.sacrifice(amount).encodeABI(),
+      },
+      GENESIS_ACCOUNT_PRIVATE_KEY
+    );
+    // console.log(`encoded(${amount}) => `, contract.methods.sacrifice(amount).encodeABI());
+
+    // send txn...
+    const txnResult = await customWeb3Request(context.web3, "eth_sendRawTransaction", [
+      tx.rawTransaction
+    ]);
+
+    // produce a block with this txn...
+    await context.createBlock({ transactions: [txnResult.result] });
+
+    // and get receipt
+    const receipt = await context.web3.eth.getTransactionReceipt(txnResult.result);
+
+    return {
+      txnResult,
+      receipt,
+    };
+
+  };
+
+  it("should be accessible from a smart contract", async function () {
```

Comment:
This test is failing because the base cost of calling the wrapper contract seems to vary slightly based on the amount of gas the precompile consumes.

Note that if I hack the `sacrifice` precompile to always consume 0 gas, the base cost is consistent.

Reply:
Mystery solved: Ethereum charges for txn data (`4` for any zero, `16` for any non-zero). See 2a5120c for more details.